api_core/google/api_core/operation.py

```diff
@@ -145,21 +145,28 @@ def _set_result_from_operation(self):
                 )
                 self.set_exception(exception)
 
-    def _refresh_and_update(self):
-        """Refresh the operation and update the result if needed."""
+    def _refresh_and_update(self, retry=polling.DEFAULT_RETRY):
+        """Refresh the operation and update the result if needed.
+
+        Args:
+            retry (google.api_core.retry.Retry): (Optional) How to retry the RPC.
+        """
         # If the currently cached operation is done, no need to make another
         # RPC as it will not change once done.
         if not self._operation.done:
-            self._operation = self._refresh()
+            self._operation = self._refresh(retry=retry)
```

Comment:
Adding `retry` into inheritor of the `PollingFuture()` (the one above) and passing it further. As `self._refresh` is `functools.partial`, we can just pass `retry` as a named arg - it'll be added to those args which were prepared while creating `functools.partial`